### DISCUSSION TITLE
[Item] Cite item with pages

### DISCUSSION THREAD

**Discussion by osservazione** [windingwind/zotero-better-notes#758] [Note Templates]
Hi all, I'm a newbie here, and I desperately need a simple template for all my non-digital content (i.e. traditional books). Indeed I'd like to have a 'child-note' template with these elements:

Name of the Note

"Quotation"

Ref. Author 2010, p.23-24

Of course, the reference is that of the 'father'.
I hope it is all clear, thanks for your efforts guys!

Greetings from Italy :)

**Comment by windingwind** [comment]
Hi. Do you want it to be generated from a single annotation (which is controlled by the QuickNote template), or from an item (which is controlled by an item template)?

![image](https://github.com/windingwind/zotero-better-notes/assets/33902321/561fbb9b-c322-4ede-b778-5dbc98324d7e)

**Reply by osservazione** [reply]
Hi! Thank you for your reply. If I'm not wrong, I need the second one: item (controlled by an item template). My problem is dealing with a large amount of old non-digitalized books (scanner them is not an option affordable) and right now I do these 3 operations:

1. click on [+]
2. search the reference
3. input the page number 

So the key point is: Is it possible to have the reference in the template so that I have only to input the page number?

Thanks a lot for your time and patience :)

![image](https://github.com/windingwind/zotero-better-notes/assets/148261236/58e4fcd2-9866-4774-a361-18527f13201a)
![image](https://github.com/windingwind/zotero-better-notes/assets/148261236/97ee71ac-9363-4feb-b359-8fc5b24f3351)
![image](https://github.com/windingwind/zotero-better-notes/assets/148261236/aea65a34-dd37-4bba-8e60-4cf5ecc6f288)

**Comment by windingwind** [comment]
so if i understand correctly, you want a template to insert the citation of the parent item of the current note?

**Reply by osservazione** [reply]
Yes, I think so. A template for manually inserting the text/quotation AND the page of a physical book (parent item)

**Reply by windingwind** [reply]
Where will the page number be from? Have a popup window to ask you to manually input every time, or leave it blank so you can input it in note editor later?

**Reply by osservazione** [reply]
both solutions would be good, but the popup window I think it would be super! In other words, at the stake is to reduce the number of the input operations

**Reply by windingwind** [reply]
Here are the steps to use:

1. Update to the latest beta (https://github.com/windingwind/zotero-better-notes/releases/tag/1.1.4-29) + Zotero 7
2. Copy the following template share code:
```yaml
# This template is specifically for importing/sharing, using better 
# notes 'import from clipboard': copy the content and
# goto Zotero menu bar, click Edit->New Template from Clipboard.  
# Do not copy-paste this to better notes template editor directly.
name: "[item] my citation"
content: |-
  ${{
  	const pages = _env.dryRun
  		? ""
  		: Zotero.getMainWindow().prompt("Cite page(s):");
  	return await Zotero.BetterNotes.api.convert.item2citation(
  		[topItem.id],
  		[{
  			label: "page",
  			locator: pages,
  		}]
  	);
  }}$
```
3. menu -> Edit -> Import template from clipboard
4. Open the reader, and create template note from the sidebar menu:
![image](https://github.com/windingwind/zotero-better-notes/assets/33902321/8ced1c89-4f46-462d-b565-77e5fe0d2b38)

5. Select the target template in the popup.
6. Input page label, or leave it blank.
7. Done

**Reply by osservazione** [reply]
First of all: thank you so much! I followed the instructions and all work but only for pdf items.

For non-pdf items, this is what happens:

![image](https://github.com/windingwind/zotero-better-notes/assets/148261236/bcf86ee0-012b-4ae1-83fb-9e0e3c7059ae)

![image](https://github.com/windingwind/zotero-better-notes/assets/148261236/8ef99bba-7a75-4479-8923-eeab4b6af4b5)

and click on the "[item] my citation" template, the result is an empty note:

![image](https://github.com/windingwind/zotero-better-notes/assets/148261236/abd52469-9734-4ebc-8065-f662c12163b3)

**Reply by windingwind** [reply]
Seems to be a bug of the latest prerelease. I've fixed in https://github.com/windingwind/zotero-better-notes/releases/tag/1.1.4-33, please try again

**Reply by osservazione** [reply]
Super! Now it works. Many thanks, I offer you a coffee 👍
